partner_store/models/res_partner.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    type = fields.Selection(selection_add=[("store", "Store Address")])
```

Comment:
I think a good improvement would be to set a fallback value (`'other'` or `'contact'`).
```suggestion
    type = fields.Selection(selection_add=[("store", "Store Address")], ondelete={"store": "other"})
```